src/components/lab/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,326 @@
+'use client';
+
+import React from 'react';
+
+import {KeyCode} from '../../../constants';
+import {useControlledState, useForkRef} from '../../../hooks';
+import {useFormResetHandler} from '../../../hooks/private';
+import {TextInput} from '../../controls/TextInput';
+import type {BaseInputControlProps} from '../../controls/types';
+import {getInputControlState} from '../../controls/utils';
+import {block} from '../../utils/cn';
+
+import {NumericArrows} from './NumericArrows/NumericArrows';
+import {
+    areStringRepresentationOfNumbersEqual,
+    clampToNearestStepValue,
+    getInputPattern,
+    getInternalState,
+    getParsedValue,
+    getPossibleNumberSubstring,
+    updateCursorPosition,
+} from './utils';
+
+import './NumberInput.scss';
+
+const b = block('number-input');
+
+export interface NumberInputProps
+    extends Omit<
+        BaseInputControlProps<HTMLInputElement>,
+        'error' | 'value' | 'defaultValue' | 'onUpdate'
+    > {
+    /** The control's html attributes */
+    controlProps?: Omit<React.InputHTMLAttributes<HTMLInputElement>, 'min' | 'max' | 'onChange'>;
+    /** Help text rendered to the left of the input node */
+    label?: string;
+    /** Indicates that the user cannot change control's value */
+    readOnly?: boolean;
+    /** User`s node rendered before label and input node */
+    startContent?: React.ReactNode;
+    /** User`s node rendered after input node and clear button */
+    endContent?: React.ReactNode;
+    /** An optional element displayed under the lower right corner of the control and sharing the place with the error container */
+    note?: React.ReactNode;
+    /**Describes the validation state */
+    validationState?: 'invalid' | undefined;
+
+    /** Hides increment/decrement buttons at the end of control
+     */
+    hiddenControls?: boolean;
+    /** min allowed value. It is used for clamping entered value to allowed range
+     * @default Number.MAX_SAFE_INTEGER
+     */
+    min?: number;
+    /** max allowed value. It is used for clamping entered value to allowed range
+     * @default Number.MIN_SAFE_INTEGER
+     */
+    max?: number;
+    /** Delta for incrementing/decrementing entered value with arrow keyboard buttons or component controls
+     * @default 1
+     */
+    step?: number;
+    /** Step multiplier when shift button is pressed
+     * @default 10
+     */
+    shiftMultiplier?: number;
+    /** Enables changing value by scrolling mousewheel on with cursor on the input
+     * @default  false
+     */
+    allowMouseWheel?: boolean;
```

Comment:
We've decided to remove the wheel behaviour completely. It turned out that currently it brings a lot of bugs and inconsistency between mouse and touchpad devices. Maybe we reimplement it in the future.